src/fields/field_constant.impl.hh

```diff
@@ -132,6 +144,13 @@ void FieldConstant<spacedim, Value>::value_list (const std::vector< Point >  &po
 }
 
 
+template <int spacedim, class Value>
+typename Value::return_type const & FieldConstant<spacedim, Value>::time_dependent_value(double time)
+{
+	return table_function_.value(time);
```

Comment:
- Introduce FieldTimeFunction.
- Make it derived from FieldConstant.
- Only implement get_input_type and set_time methods.
- Move specific code from init_from_input into set_time (possible also for other Fields).